root/static/scripts/common/utility/isDateEmpty.js

```diff
@@ -31,7 +31,7 @@ export function isDateObservableEmpty(
 
 export default function isDateEmpty(
   date: ?PartialDateT | ?PartialDateStringsT,
-): boolean {
+): boolean %checks {
```

Comment:
Remind me, what did this do?

Reply:
The documentation is hard to find, but - https://flow.org/en/docs/types/functions/#toc-predicate-functions

Basically it preserves any refinements that are made inside the function to apply at the call site (but they have to be very simple checks for this to work).